api/notd/collection_processor.py

```diff
@@ -0,0 +1,69 @@
+import asyncio
+import json
+import logging
+
+from core.exceptions import BadRequestException, NotFoundException
+from core.requester import Requester
+from core.requester import ResponseException
+from core.web3.eth_client import EthClientInterface
+
+from notd.model import RetrievedCollection
+
+
+class CollectionDoesNotExist(NotFoundException):
+    pass
+
+class CollectionProcessor:
+    def __init__(self, requester: Requester, ethClient: EthClientInterface):
+        self.requester = requester
+        self.ethClient = ethClient
+        with open('./contracts/IERC721Metadata.json') as contractJsonFile:
+            erc721MetdataContractJson = json.load(contractJsonFile)
+        self.erc721MetdataContractAbi = erc721MetdataContractJson['abi']
+        self.erc721MetadataNameFunctionAbi = [internalAbi for internalAbi in self.erc721MetdataContractAbi if internalAbi['name'] == 'name'][0]
+        self.erc721MetadataSymbolFunctionAbi = [internalAbi for internalAbi in self.erc721MetdataContractAbi if internalAbi['name'] == 'symbol'][0]
+
+
+    async def retrieve_collection(self, address: str) -> RetrievedCollection:
+        response = 0
+        retryCount = 0
+        try:
+            tokenMetadataNameResponse = await self.ethClient.call_function(toAddress=address, contractAbi=self.erc721MetdataContractAbi, functionAbi=self.erc721MetadataNameFunctionAbi)
+        except BadRequestException as exception:
+            tokenMetadataNameResponse = [None]
```

Comment:
why is this in an array?

Reply:
ah the eth responses are always an array - this makes me question how the symbol one could ever work - you arent un-indexing the array response anywhere.

Reply:
it would be better to do `tokenMetadataName = tokenMetadataNameResponse[0]` above and assign `tokenMetadataName = None here instead of doing the raw value`